src/main/java/com/google/gcp_variant_transforms/library/SchemaGeneratorImpl.java

```diff
@@ -23,59 +26,163 @@ public TableSchema getSchema(VCFHeader vcfHeader) {
+          .setName(fieldName)
+          .setDescription(SchemaUtils.callSubFieldNameToDescriptionMap.get(fieldName))
+          .setMode(SchemaUtils.callSubFieldNameToModeMap.get(fieldName))
+          .setType(SchemaUtils.callSubFieldNameToTypeMap.get(fieldName));
       callSubFieldsBuilder.add(callSubField);
     }
+    // Adds the FORMAT fields under the Calls record
+    for (VCFFormatHeaderLine formatHeaderLine : vcfHeader.getFormatHeaderLines()){
+        TableFieldSchema formatField = new TableFieldSchema()
+            .setName(getSanitizedFieldName(formatHeaderLine.getID()))
+            .setDescription(formatHeaderLine.getDescription())
+            .setMode(SchemaUtils.BQFieldMode.NULLABLE) // Always NULLABLE
+            .setType(SchemaUtils.HTSJDKTypeToBQTypeMap.get(formatHeaderLine.getType()));
+        callSubFieldsBuilder.add(formatField);
+      }
     return callSubFieldsBuilder.build();
   }
+
+  /**
+   * Creates and returns an Alternate Base record's subfields.
+   * @param vcfHeader
+   * @return a list of alternate base sub-fields
+   */
+  @VisibleForTesting
+  protected ImmutableList<TableFieldSchema> getAltBaseSubFields(VCFHeader vcfHeader) {
+    ImmutableList.Builder<TableFieldSchema> altBaseSubFieldsBuilder = new ImmutableList.Builder<TableFieldSchema>();
+    // Adds the alternate bases allele column.
+    TableFieldSchema altSubField = new TableFieldSchema()
+        .setName(Constants.ColumnKeyNames.ALTERNATE_BASES_ALT)
+        .setDescription(SchemaUtils.FieldDescription.ALTERNATE_BASES_ALT)
+        .setMode(SchemaUtils.BQFieldMode.NULLABLE)
+        .setType(SchemaUtils.BQFieldType.STRING);
+    altBaseSubFieldsBuilder.add(altSubField);
+    // Adds the INFO fields with Number = A (i.e., one value for each alternate) among alternates.
+    for (VCFInfoHeaderLine infoHeaderLine : vcfHeader.getInfoHeaderLines()){
+      if (infoHeaderLine.getCountType() == VCFHeaderLineCount.A){
+        TableFieldSchema infoSubField = new TableFieldSchema()
+            .setName(getSanitizedFieldName(infoHeaderLine.getID()))
+            .setDescription(infoHeaderLine.getDescription())
+            .setMode(SchemaUtils.BQFieldMode.NULLABLE) // Always NULLABLE
+            .setType(SchemaUtils.HTSJDKTypeToBQTypeMap.get(infoHeaderLine.getType()));
+        altBaseSubFieldsBuilder.add(infoSubField);
+      }
+    }
+    return altBaseSubFieldsBuilder.build();
+  }
+
+  /**
+   * Returns the sanitized field name according to BigQuery restrictions.
+   *
+   * BigQuery field names must follow `[a-zA-Z][a-zA-Z0-9_]*`. This method
+   * converts any unsupported characters to an underscore. Also, if the first
+   * character does not match `[a-zA-z]`, it prepends `FALLBACK_FIELD_NAME_PREFIX`
+   * to the name.
+   * @param fieldName: Name of the field to sanitize
+   * @return sanitized field name
+   */
+  protected String getSanitizedFieldName(String fieldName){
```

Comment:
As discussed, this probably should be moved.

Reply:
Moved this function to SchemaUtils.java after discussing with Saman and Zhiqiang